src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/handlers/CancellationCaseEventHandler.java

```diff
@@ -0,0 +1,201 @@
+            .forEach(cancellationEvaluateResponse -> {
+                DmnValue<String> taskCategories = cancellationEvaluateResponse.getTaskCategories();
+                DmnValue<String> processCategories = cancellationEvaluateResponse.getProcessCategories();
+                sendCancellationMessage(
+                    eventInformation.getCaseId(),
+                    taskCategories,
+                    processCategories
+                );
+            });
+    }
+
+    private EvaluateDmnRequest buildEvaluateDmnRequest(
+        String previousStateId,
+        String eventId,
+        String newStateId
+    ) {
+        Map<String, DmnValue<?>> variables = Map.of(
+            "event", dmnStringValue(eventId),
+            "state", dmnStringValue(newStateId),
+            "fromState", dmnStringValue(previousStateId)
+        );
+
+        return new EvaluateDmnRequest(variables);
+    }
+
+    private void sendCancellationMessage(String caseReference,
+                                         DmnValue<String> categories,
+                                         DmnValue<String> processCategories) {
+        Set<SendMessageRequest> cancellationMessageRequests =
+            buildCancellationMessageRequest(caseReference, categories, processCategories);
+
+        cancellationMessageRequests.forEach(message -> {
+                if (message != null) {
+                    workflowApiClient.sendMessage(serviceAuthGenerator.generate(), message);
+                }
+            }
+        );
+
+    }
+
+    private Set<SendMessageRequest> buildCancellationMessageRequest(
+        String caseReference,
+        DmnValue<String> categories,
+        DmnValue<String> processCategories
+    ) {
+
+        SendMessageRequest oldFormatCancellationMessage =
+            createOldFormatCancellationMessage(caseReference, categories, processCategories);
+        SendMessageRequest cancellationMessage =
+            createCancellationMessage(caseReference, categories, processCategories);
+
+        return new HashSet<>(asList(oldFormatCancellationMessage, cancellationMessage));
+    }
+
+    /**
+     * This method creates a SendMessageRequest to cancel all tasks matching a set of correlation keys.
+     * This method supports multiple categories.
+     *
+     * @param caseReference the case id to be used as a correlation key
+     * @param categories    the categories to be used as correlation keys
+     * @return The message request object.
+     */
+    private SendMessageRequest createCancellationMessage(String caseReference,
+                                                         DmnValue<String> categories,
+                                                         DmnValue<String> processCategories) {
+
+        Map<String, DmnValue<?>> correlationKeys = new HashMap<>();
+        correlationKeys.put("caseId", dmnStringValue(caseReference));
+
+        if (categories != null && categories.getValue() != null) {
```

Comment:
suggestion: This block and the block below has the same implementation. If you select this block and refactor it into a method you can then re-use that method for the block below.